bootstrapWorktree BOOTSTRAP_CONFIGS omits gitlab-workflow config

## Context

Surfaced during #9491 / PR #12792 whitebox-e2e in a fresh `.claude/worktrees/<name>/` worktree. The webpack dev server failed to compile:

```
Error: Cannot find module '.../ai/mcp/server/gitlab-workflow/config.mjs' imported from .../ai/services/gitlab-workflow/GitLabClient.mjs
```

Running `bootstrapWorktree({mainCheckout, projectRoot})` hydrated `ai/config.mjs` + the `github-workflow` / `knowledge-base` / `memory-core` / `neural-link` server configs (all `skip (exists)` / copied), but **never touched `gitlab-workflow`** — its `config.mjs` stayed missing.

## The Problem

#10095 (CLOSED) fixed the *general* "worktrees fail on gitignored `ai/mcp/server/*/config.mjs`" bootstrap problem, and #10351 (CLOSED) added the npm-install / `--build-all` path. But the **`gitlab-workflow` MCP server was added later** (GitLab track) and was **never added to `BOOTSTRAP_CONFIGS`**. So a fresh worktree that imports the gitlab service chain hits the exact #10095 `ERR_MODULE_NOT_FOUND` symptom — for a server #10095's fix doesn't enumerate. The config list is a hand-maintained allow-list that drifts every time a server is added.

## The Architectural Reality

- `ai/scripts/migrations/bootstrapWorktree.mjs` — `BOOTSTRAP_CONFIGS` (~line 119) is the explicit list of `ai/mcp/server/*/config.mjs` files to hydrate. It enumerates github-workflow / knowledge-base / memory-core / neural-link but **not** gitlab-workflow.
- `ai/mcp/server/gitlab-workflow/config.template.mjs` exists; `config.mjs` is gitignored (per the standard per-server override pattern).

## The Fix

Add `ai/mcp/server/gitlab-workflow/config.mjs` to `BOOTSTRAP_CONFIGS` (mirroring the sibling entries). **Stronger fix to prevent recurrence:** auto-derive the list by globbing `ai/mcp/server/*/config.template.mjs` so newly-added servers are hydrated automatically — eliminating the drift class, not just this instance.

## Decision Record impact

`none`. Related: #10095 (CLOSED — general gitignored-config bootstrap mechanism), #10351 (CLOSED — npm-install / `--build-all` for test-ready worktrees).

## Acceptance Criteria

- [ ] `BOOTSTRAP_CONFIGS` includes gitlab-workflow (or the list auto-derives from the server dirs).
- [ ] A fresh worktree bootstrapped via `bootstrapWorktree` has `ai/mcp/server/gitlab-workflow/config.mjs` hydrated; importing the gitlab service chain no longer throws `ERR_MODULE_NOT_FOUND`.
- [ ] (Preferred) the list is derived from `ai/mcp/server/*/` so future server additions can't reintroduce this drift.

## Out of Scope

- The npm-install / `--build-all` path (#10351, done) and theme CSS build (covered by `--build-all`).

Origin Session ID: 31325e4f-0fdf-4627-96ed-7983dd1b1002

Retrieval Hint: "bootstrapWorktree BOOTSTRAP_CONFIGS gitlab-workflow missing worktree ERR_MODULE_NOT_FOUND".

Authored by Claude Opus 4.8 (Claude Code), @neo-opus-ada.

--- a/ai/scripts/migrations/bootstrapWorktree.mjs
+++ b/ai/scripts/migrations/bootstrapWorktree.mjs
@@ -108,6 +108,7 @@
  */
 import {execFile}       from 'child_process';
 import fs               from 'fs/promises';
+import {existsSync, readdirSync} from 'fs';
 import path             from 'path';
 import {fileURLToPath}  from 'url';
 import {promisify}      from 'util';
@@ -116,12 +117,28 @@ import {materializeServerConfigTemplate} from '../setup/initServerConfigs.mjs';
 
 const execFileAsync = promisify(execFile);
 
+/**
+ * The set of gitignored config overlays a fresh worktree must hydrate from the main
+ * checkout: the Tier-1 `ai/config.mjs` plus each per-server `ai/mcp/server/<name>/config.mjs`.
+ *
+ * **Auto-derived, not hand-maintained.** The per-server list is globbed at load time from the
+ * `ai/mcp/server/*` dirs that ship a `config.template.mjs` — the same predicate
+ * `initServerConfigs` uses. The previous hand-maintained allow-list silently drifted whenever a
+ * new MCP server was added: the new server's gitignored `config.mjs` went un-hydrated, so a
+ * worktree importing that server's chain crashed with `ERR_MODULE_NOT_FOUND`. Deriving from the
+ * filesystem eliminates that drift class — any server shipping a template is hydrated
+ * automatically; dirs without one (e.g. `file-system`, `shared`) are excluded.
+ */
+const serverConfigsDir = path.resolve(path.dirname(fileURLToPath(import.meta.url)), '../../mcp/server');
+
 export const BOOTSTRAP_CONFIGS = [
     'ai/config.mjs',
-    'ai/mcp/server/github-workflow/config.mjs',
-    'ai/mcp/server/knowledge-base/config.mjs',
-    'ai/mcp/server/memory-core/config.mjs',
-    'ai/mcp/server/neural-link/config.mjs'
+    ...readdirSync(serverConfigsDir, {withFileTypes: true})
+        .filter(entry => entry.isDirectory())
+        .map(entry => entry.name)
+        .filter(name => existsSync(path.join(serverConfigsDir, name, 'config.template.mjs')))
+        .sort()
+        .map(name => `ai/mcp/server/${name}/config.mjs`)
 ];
 
 /**

--- a/test/playwright/unit/ai/scripts/migrations/bootstrapWorktree.spec.mjs
+++ b/test/playwright/unit/ai/scripts/migrations/bootstrapWorktree.spec.mjs
@@ -87,8 +87,25 @@ test.describe('ai/scripts/bootstrapWorktree', () => {
         }
     });
 
-    test('exports the canonical BOOTSTRAP_CONFIGS list', () => {
-        expect(BOOTSTRAP_CONFIGS).toEqual(fixtureConfigs);
+    test('auto-derives BOOTSTRAP_CONFIGS from server dirs that ship a config.template.mjs', () => {
+        // Tier-1 operator overlay is always first.
+        expect(BOOTSTRAP_CONFIGS[0]).toBe('ai/config.mjs');
+
+        // Every MCP server that ships a config.template.mjs is hydrated — including
+        // gitlab-workflow, whose omission from the old hand-maintained list broke fresh
+        // worktrees with ERR_MODULE_NOT_FOUND (the regression this guards against).
+        for (const name of ['github-workflow', 'gitlab-workflow', 'knowledge-base', 'memory-core', 'neural-link']) {
+            expect(BOOTSTRAP_CONFIGS).toContain(`ai/mcp/server/${name}/config.mjs`);
+        }
+
+        // Dirs without a config.template.mjs (file-system, shared) are excluded.
+        expect(BOOTSTRAP_CONFIGS).not.toContain('ai/mcp/server/file-system/config.mjs');
+        expect(BOOTSTRAP_CONFIGS).not.toContain('ai/mcp/server/shared/config.mjs');
+
+        // Every entry is the Tier-1 overlay or a well-formed per-server overlay path.
+        for (const rel of BOOTSTRAP_CONFIGS) {
+            expect(rel === 'ai/config.mjs' || /^ai\/mcp\/server\/[^/]+\/config\.mjs$/.test(rel)).toBe(true);
+        }
     });
 
     test('resolveCliProjectRoot climbs migrations/ → scripts/ → ai/ → root (#12147)', () => {

